glommio/src/executor/mod.rs

```diff
@@ -1054,23 +1065,35 @@ impl LocalExecutor {
                     now
                 };
 
-                let mut tasks_executed_this_loop = 0;
-                loop {
-                    let mut queue_ref = queue.borrow_mut();
-                    if self.need_preempt() || queue_ref.yielded() {
-                        break;
-                    }
+                let (runtime, tasks_executed_this_loop) = {
+                    #[cfg(feature = "stall-detection")]
+                    let _stall_detector_guard = self
+                        .stall_detector
+                        .enter_task_queue(
+                            queue.borrow_mut().name.clone(),
+                            time,
+                            self.preempt_timer_duration(),
+                        )
+                        .unwrap();
+
+                    let mut tasks_executed_this_loop = 0;
+                    loop {
+                        let mut queue_ref = queue.borrow_mut();
+                        if self.need_preempt() || queue_ref.yielded() {
+                            break;
+                        }
 
-                    if let Some(r) = queue_ref.get_task() {
-                        drop(queue_ref);
-                        r.run();
-                        tasks_executed_this_loop += 1;
-                    } else {
-                        break;
+                        if let Some(r) = queue_ref.get_task() {
+                            drop(queue_ref);
+                            r.run();
+                            tasks_executed_this_loop += 1;
+                        } else {
+                            break;
+                        }
                     }
-                }
 
-                let runtime = time.elapsed();
+                    (time.elapsed(), tasks_executed_this_loop)
```

Comment:
I don't mind doing RAII for the stall detector, but then it should be dropped manually as soon as we exit the loop, and not at the end of the function. You don't want to include accounting time in the backtrace